envoy-filters/mgw-source/filters/http/mgw-wasm-websocket/filter.cc

```diff
@@ -86,24 +91,70 @@ void MgwWebSocketContext::onCreate() {
 FilterHeadersStatus MgwWebSocketContext::onRequestHeaders(uint32_t, bool) {
   LOG_TRACE(std::string("onRequestHeaders called mgw_WASM_websocket") + std::to_string(id()));
 
-  // Create a new gRPC bidirectional stream.
-  establishNewStream();
-
+  auto requestHeaderResult = getRequestHeaderPairs();
+  auto headerPairs = requestHeaderResult->pairs();
+  for (auto& p : headerPairs) {
+    if (std::string(p.first) == X_REQUEST_ID) {
+      this->x_request_id_ = std::string(p.second);
+    }
+  }
   // Extract ext_authz dynamic metadata and assign it to a member variable 
   auto buffer = getProperty<std::string>({"metadata", "filter_metadata", "envoy.filters.http.ext_authz"});
   if (buffer.has_value() && buffer.value()->size() != 0) {
     auto pairs = buffer.value()->pairs();
     for (auto &p : pairs) {
+      // TODO (thushani) Implement fixes for throttling
```

Comment:
shall we omit this logic from here? It feels like we need an entirely new way to handle throttling as now enforcer does not perform throttling .